### ISSUE TITLE
Log presentation

### ISSUE THREAD

**Issue by ThomDietrich** [openhab/openhab-vscode#9]
Present logging data inside the IDE.

* Access via openhab.log/events.log, Karaf Remote Console or via a dedicated connection (I've got experience with openHAB logging via [log4j over TCP](https://blog.lanyonm.org/articles/2015/12/29/log-aggregation-log4j-spring-logstash.html) sent to elasticsearch)?
* Presentation as raw text or as interactive panel?
* Integration of e.g. #8

**Comment by ThomDietrich** [comment]
@kaikreuzer following along the idea in #7 to exploit the REST api, do you see any possibility to use it for log retrieveal? An important feature in relation to the above. It would be desirable to get the log in a notated format - which is currently not available to the public.

**Comment by MHerbst** [comment]
Having access to the logs via REST would also allow remote access to the log information via myopenhab.

**Comment by kaikreuzer** [comment]
To me this sounds not like a desirable feature for the VS Code extension. VS Code should imho be used as an editor, not as a remote management console... And logs are not edited, so they don't really fit into the main use case here.
Wouldn't a webconsole be a much much better alternative? Or am I missing something?

**Comment by ThomDietrich** [comment]
I should explain in for detail. Of course I'm not talking about editing the logs. This ticket is aimed at showing the log in a view next to the [output/debug pane](https://4.bp.blogspot.com/-St5URa61wE8/WNAOn7qnBxI/AAAAAAAABQQ/QpnrTKxxhFU75yzGVlojRhRBIxURyN7iwCLcB/s1600/Screen%2BShot%2B2017-03-20%2Bat%2B18.16.26.png)
I believe seeing the output of the execution of rules and incoming events on the event bus is needed during set up and rule development.

Now of course it would be desirable to only see output related to the currently edited rule for example. That's why I've emphasized "notated format" above :smiley:

Wdyt?

**Comment by kubawolanin** [comment]
@ThomDietrich we may as well display openHAB Log Viewer / [frontail](https://github.com/mthenw/frontail), but it's too tightly coupled with [openHABian](http://docs.openhab.org/installation/openhabian.html) at the moment.

The side view is basically a browser embedded inside IDE. We can display whatever we want here :)
Although I'd first think of a webconsole @kaikreuzer mentioned, e.g. a separate openHAB UI aggregating logs. Maybe a fork of frontail bundled as a openHAB UI add-on? Just suggesting the ideas... 😃 

![](https://github.com/openhab/openhab-vscode/raw/master/images/openhab-sitemap.gif)

**Comment by ThomDietrich** [comment]
The "openHAB Log Viewer"/frontail is not meant to be a real solution, it's just a tail on a webpage ;)
To store, filter and query logs I've got a prototype solution based on logstash+elasticsearch+kibana working - does the job quite nicely.

I followed the discussion around the web console [here](https://community.openhab.org/t/log-manager-in-paperui/29175) and do agree that such a UI would be interesting in some cases.

Maybe I should clarify for everyone still confused that when talking about the "log" I'm talking about the integration of information generated by rules and logged due to event bus changes in meaningful ways. At first that could happen as a simple "Log View" but in the long run I'd strive for a more specialized integration.

Let me define a few user stories to make the idea clearer:

* As the end user I want a powerful editor to help build my rules (and other files). (Check)
* As the end user I want to test my rules by simulating/injecting events (e.g. "FrontDoor changed to open") from inside the editor to evaluate their correct behavior. (See #8)
* As the end user I want to see the result of the rule execution inside the editor to evaluate their correct behavior. (*this issue*)
* As the end user I want to know the current state of an item from inside the editor to have a better understanding of my setup and of the behavior of rules. (Not yet discussed)
